.github/actions/perform_tests/action.yml

```
@@ -0,0 +1,31 @@
+name: 'Perform Tests'
+description: 'This action contains all the usual steps needed to perform the unit tests'
+
+inputs:
+  needs-shared-tests:
+    description: 'Flag to determine whether the action needs to perform the Shared  module tests'
+    required: true
+  needs-storage-tests:
+    description: 'Flag to determine whether the action needs to perform the Storage module tests'
+    required: true
+  needs-client-tests:
+    description: 'Flag to determine whether the action needs to perform the Client module tests'
+    required: true
+  github-token:
+    description: 'The Github Token'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Run Shared tests if needed
+      shell: bash
+      run: |
+        bundles_under_testing=$(./.github/scripts/process_test_bundles.sh ${{ inputs.needs-shared-tests }} ${{ inputs.needs-storage-tests }} ${{ inputs.needs-client-tests }})
```

I guess the logic is very much depending on the order of the booleans here (and it being in sync with the order of the strings in the scripts). Not a problem at the moment, just think this is a potential point of failure if and when updating this 🤔 

Not even sure if it would work, but how about trying to inline it like:
```suggestion
        bundles_under_testing="$(${{ inputs.needs-shared-tests }} && echo "SharedTests") $(${{ inputs.needs-storage-tests }} && echo "StorageTests") $(${{ inputs.needs-client-tests }} && echo "ClientTests")"
```

I tested with boolean values instead of the substitution on my terminal and it seems to work
```
echo "$(false && echo "SharedTests") $(true && echo "StorageTests") $(true && echo "ClientTests")"
output> StorageTests ClientTests
```

If it works, it would even make you not need the `process_test_bundles.sh` script anymore. Wdyt?

---

Nice idea!
The outcome input though needs a comma-separated string that the Fastlane action takes and creates the array of test bundles to execute.
it may be a bit trickier to add it to the inline (I guess), that's the main reason I managed to make a script out of it.
Easier to change and fewer chances to compromise the command in the run step.
In case of adding /  removing test bundles, it will be easier to add/remove a `${{ inputs.needs-test-bundle }}` and play with the script.

---

Understood, I didn't pay attention to the comma before 😅 It is indeed a bit trickier to inline in that case since you don't know what comes before or after 🤔 

I'm guessing if you pass like "SharedTests,StorageTests," with a trailing comma at the end it fails, right? Because if that works you could just add a comma at the end of the echos and remove the spacing in between and my suggestion would still be valid I guess:
```suggestion
        bundles_under_testing="$(${{ inputs.needs-shared-tests }} && echo "SharedTests,")$(${{ inputs.needs-storage-tests }} && echo "StorageTests,")$(${{ inputs.needs-client-tests }} && echo "ClientTests,")"
```

In case that doesn't work, also no problem at all leaving it like it is since it works. My only fear was that (from what I understood) it depends too much on the order on both the arguments and the strings on the script, which is a bit hidden and can be a point of failure in the future.

---

Thanks for the suggestion 🙏 
Haven't tried with the coma at the end as I was focusing on getting to the cleanest possible solution.
But I get your point, totally!
Can try with the suggestion and trigger it - otherwise revert the commit 👌